tests/integration/test_bayesian_optimization.py

```diff
@@ -281,7 +281,7 @@ def build_model(data: Dataset) -> GPflowPredictor:
         with tensorboard_writer(summary_writer):
 
             dataset = (
-                BayesianOptimizer(observer, search_space)
+                BayesianOptimizer(observer, search_space)  # type: ignore
```

Comment:
I think this is due to the different types of acquisition rules (some with state some without) confusing mypy.